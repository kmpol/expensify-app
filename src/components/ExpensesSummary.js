import React from 'react'
import { connect } from 'react-redux';
import selectedExpenses from '../selectors/expenses'
import selectedExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component {

    render() {
        const grammarExpenseDisplay = this.props.expenseCount === 1 ? "expense" : "expenses"
        return (
            <div>
                <p>Viewing {this.props.expenseCount} {grammarExpenseDisplay} with total: ${numeral(this.props.expensesInTotal / 100).format('0.00')}</p>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectedExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesInTotal: selectedExpensesTotal(visibleExpenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary)