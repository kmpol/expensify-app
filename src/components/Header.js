import { NavLink } from 'react-router-dom'
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux'

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink activeClassName="is-active" to="/dashboard">Home page</NavLink>
        <NavLink activeClassName="is-active" to="/create">Create expense</NavLink>
        <NavLink activeClassName="is-active" to="/help">Help page</NavLink>
        <button onClick={startLogout()}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout)
    }
}

export default connect(undefined, mapDispatchToProps)(Header)