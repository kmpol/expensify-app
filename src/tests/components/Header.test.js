import React from 'react'
import toJSON from 'enzyme-to-json'
import { Header } from '../../components/Header'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });
test('Should render Header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => { }} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('Should call startLogout on button click', () => {
    const startLogoutSpy = jest.fn()
    const wrapper = shallow(<Header startLogout={startLogoutSpy} />)
    wrapper.find('button').simulate('click')
    expect(startLogoutSpy).toHaveBeenCalled()
})