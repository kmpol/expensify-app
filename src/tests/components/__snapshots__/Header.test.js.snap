// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    to="/dashboard"
  >
    Home page
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help page
  </NavLink>
  <button>
    Logout
  </button>
</header>
`;
